### DISCUSSION TITLE
Installing on Heroku Free Tier

### DISCUSSION THREAD

**Discussion by ched-dev** [directus/directus#12527] [Archive]
I have pushed up my site to Heroku's free tier (free dynos) with the Heroku Postgres add-on but when my `directus start` runs, it can't connect to the database.

I get the error:
> error: no pg_hba.conf entry for host "IP_ADDRESS", user "DB_USER", database "DB_NAME", no encryption

When searching, I found that [Heroku requires all production databases to use SSL](https://help.heroku.com/DR0TTWWD/seeing-fatal-no-pg_hba-conf-entry-errors-in-postgres). So I set the env var `DB_SSL="true"` and get the following error:

> Error: self signed certificate

Some more searching brought me to more [Heroku documentation](https://devcenter.heroku.com/articles/connecting-heroku-postgres#connecting-in-node-js) telling me to set `PGSSLMODE=no-verify` which I did. Unfortunately, I still get the error.

On the same documentation page it showed an example connecting with some settings:

```js
const client = new Client({
  connectionString: process.env.DATABASE_URL,
  ssl: {
    rejectUnauthorized: false
  }
});
```

So I tried to see if the directus code base has a reference to `rejectUnauthorized` and found [config options](https://docs.directus.io/configuration/config-options/#configuration-options) referencing these settings:

```
DB_CLIENT="pg"
DB_CONNECTION_STRING="postgresql://postgres:example@127.0.0.1"
DB_SSL__REJECT_UNAUTHORIZED="false"

{
	client: "pg",
	connectionString: "postgresql://postgres:example@127.0.0.1",
	ssl: {
		rejectUnauthorized: false
	}
}
```

So I added the env vars `DB_CONNECTION_STRING` and `DB_SSL__REJECT_UNAUTHORIZED="false"` and...

It's alive!

_Answered my own question to leave this here as reference for anyone else in the future_

**Comment by ched-dev** [comment]
TL;DR

On Heroku:
- Set env var `DB_SSL="true"`
- Set env var `DB_CONNECTION_STRING="postgresql://postgres:example@127.0.0.1"`
- Set env var `DB_SSL__REJECT_UNAUTHORIZED="false"`

**Comment by jennifersarahpaul** [comment]
This didn't work for me.
